sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PartitioningUtils.scala

```diff
@@ -350,7 +350,12 @@ object PartitioningUtils {
    */
   def getPathFragment(spec: TablePartitionSpec, partitionSchema: StructType): String = {
     partitionSchema.map { field =>
-      escapePathName(field.name) + "=" + escapePathName(spec(field.name))
+      val value = if (spec(field.name) == null || spec(field.name).isEmpty) {
+        DEFAULT_PARTITION_NAME
+      } else {
+        escapePathName(spec(field.name))
```

Comment:
Probably, look up time to `spec` is not big deal but I would store `spec(field.name)` to a val.